continuedev/continue
# Fix missing code snippets due to uncaught MiniSearch exception

## Description

Fixes an issue where `MiniSearch` could throw an uncaught exception during `loadSubmenuItems` in `useSubmenuContextProviders`, causing no code search results to be added to the `MiniSearch`, and code snippets to appear missing in the UI.

The issue stems from lax database constraints, allowing us to insert duplicate `code_snippets` and `code_snippet_tags` during indexing. I've added unique constraints to both tables to prevent application errors in the future.

## Checklist


## Screenshots

<img width="397" alt="image" src="https://github.com/user-attachments/assets/157c705f-829b-4130-88c5-b5272d5c48c5">

## Testing
1. Add a duplicate `code_snippet_tags` row in the code_snippets database or index a repository with duplicate `code_snippet_tags` entries.
2. Load the extension and navigate to the @Code submenu item.
3. All indexed code snippets should be visible.
